### PR TITLE
Returning the focus to the original receiver

### LINKED ISSUE #8256: Returning the focus to the original location?
@Patowhiz I hope this mightbe a simple change that is automatically applied to all dialogues.  That is, if we agree.  Here is the boxplot dialogue completed:
image

When you go in the first time, the focus - so coloured - is on the Y variable.  Filling it in usually puts the focus as shown above - so in a place rarely used.

When I return it is usually either to change what is plotted - so the original variable with the focus.  Or it is to change some of the options, in which case I don't care where the focus is.

With beginners - one of my pre-occupations now - I suggest it would be simpler if the focus was always on the original first receiver, every time the dialogue is re-opened.

If agreed, then could this be a "rule" and programmed just in one place - or do we have to tweak each dialogue?

### PR DESCRIPTION
**Pull request by N-thony**
Fixes #8256
@rdstern I have tested this with some dialogues in the Describe menu, have a look

### CODE REVIEW

**Review by rdstern — COMMENTED** [review]
@N-thony it seems to work fine on some dialogs.  Others resturn and nothing is in turquoise.  Here is an example:

image

The stem and leaf is another like that:

image

**Review by rdstern — APPROVED** [review]
Looks fine.  Perhaps @Patowhiz can confirm

File: instat/ucrSelector.vb [hunk 1]
```diff
@@ -63,6 +63,12 @@ Public Class ucrSelector
         'always load selector contents on load event because contents may have been changed at R level 
         'and the control needs to refresh the data frame names.
         LoadList()
+        'always return the focus to the first Receiver when re-opening the dialogue.
+        If lstOrderedReceivers.Count > 0 Then
+            Dim lstVisibleReceivers = lstOrderedReceivers.Where(Function(ctrl) ctrl.Visible).ToList()
```
Comment (Patowhiz) [inline instat/ucrSelector.vb:68]: @N-thony could you specify the type of `lstVisibleReceivers` explicitly. This has many merits relating to long term code maintenance.

**Review by Patowhiz — CHANGES_REQUESTED** [review]

**Comment by Patowhiz** [comment]
@rdstern I hope you have tested this with dialogs that have multiple features. I'm particularly interested with dialogs that have **multiple features** yet use **different receivers**. The sequence of reload may cause regression in the changes introduced here.

File: instat/ucrSelector.vb (hunk 1 above)
Comment (Patowhiz) [reply instat/ucrSelector.vb:68]: @N-thony I understand what the predicate is doing in the where clause.
Could you then specify the type explicitly, for example `Dim lstVisibleReceivers As List(Of Control)`. I understand that VB.Net allows dynamic types, but to be consistent with the code base and also for self documenting code, I strongly recommend static typing in this case.

**Review by rdstern — DISMISSED** [review]
@Patowhiz I have checked again.  I really like this improvement and did check with many dialogs.  I can't see any problems.  My nervousness is that you know of some problems and are really just trying to teach me the lesson that I'm not checking carefully enough?
If that's not the case, then please continue with your checks and then merge!

**Review by rdstern — APPROVED** [review]

**Comment by Patowhiz** [comment]
@rdstern thanks for your response. I'm glad you were able to check again on the dialogs that I was concerned about. 
I'm still not sure whether this change may affect performance of dialog reloads or not when it comes to wide data sets. 
@N-thony thanks for the change, would be useful to test this with dialogs that have different receivers that are visible in different options/features when using wide data sets. Nevertheless, I'm merging this.
@rdstern keep testing this feature in other PR's as well, I'll do the same.

**Review by Patowhiz — APPROVED** [review]

**Pull request merged**